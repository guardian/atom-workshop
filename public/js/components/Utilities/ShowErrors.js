import React, {PropTypes} from 'react';
import {errorPropType} from '../../constants/errorPropType';

export default class ShowErrors extends React.Component {

  static propTypes = {
    errors: PropTypes.arrayOf(errorPropType)
  }

  renderError = (error, i) => {
    return (
      <div key={i} className="form__message">
        <p className="form__message__text form__message__text--error">Error: {error.title}</p>
        <p className="form__message__text form__message__text--error">Message: {error.message}</p>
      </div>
    );
  }

  render() {
    return (
      <div>
        {this.props.errors && this.props.errors.length ? this.props.errors.map((error, i) => this.renderError(error, i)) : false}
      </div>
    );
  }
}
