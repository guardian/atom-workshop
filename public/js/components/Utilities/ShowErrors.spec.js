import React from 'react';
import ShowErrors from './ShowErrors';
import renderer from 'react-test-renderer';

let errors = [
  {
    title: 'test',
    message: 'This is a test'
  },
  {
    title: 'test2',
    message: 'This is a test'
  },
];

test('it should render', () => {
  const component = renderer.create(
    <ShowErrors errors={errors} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
