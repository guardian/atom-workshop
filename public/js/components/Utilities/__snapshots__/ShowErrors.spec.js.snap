exports[`test it should render 1`] = `
<div>
  <div
    className="form__message">
    <p
      className="form__message__text form__message__text--error">
      Error: 
      test
    </p>
    <p
      className="form__message__text form__message__text--error">
      Message: 
      This is a test
    </p>
  </div>
  <div
    className="form__message">
    <p
      className="form__message__text form__message__text--error">
      Error: 
      test2
    </p>
    <p
      className="form__message__text form__message__text--error">
      Message: 
      This is a test
    </p>
  </div>
</div>
`;
